+siibra/+core/Region.m

```diff
@@ -0,0 +1,102 @@
+                        space_ids(end +1) = dataset_specs{i, 1}.space_id;
+                        space_urls(end +1) = dataset_specs{i, 1}.url;
+                    elseif isfield(dataset_specs(i), "space_id")
+                        space_ids(end +1) = dataset_specs(i).space_id;
+                        space_urls(end +1) = dataset_specs(i).url;
+                    end
+                end
+            end
+            for region_index = 1:numel(space_ids)
+                for parcellation_index = 1:numel(parcellation.spaces)
+                    if space_ids(region_index) == parcellation.spaces(parcellation_index).id
+                        spaceAndRegion.spaces(end + 1) = parcellation.spaces(parcellation_index);
+                        spaceAndRegion.url(end + 1) = space_urls(region_index);
+                        break
+                    end
+                end
+            end
+            region.spaceAndRegionUrl = spaceAndRegion;
+        end
+
+        function spaces = get.spaces(obj)
+            spaces = obj.spaceAndRegionUrl.spaces;
+        end
+        function space = space(obj, spaceName)
+            spaceNames = {obj.spaceAndRegionUrl.spaces.name};
+            spaceIndex = siibra.internal.fuzzyMatching(spaceName, spaceNames);
+            space = obj.spaceAndRegionUrl.spaces(spaceIndex);
+        end
+        function children = get.children(obj)
+            children = obj.parcellation.getChildrenRegions(obj.name);
+        end
+        function parent = get.parent(obj)
+            parent = obj.parcellation.getParentRegion(obj.name);
+        end
+        function volume = visualizeProbabilityMapInTemplate(obj, space_name)
+            % Combine the probability map of the region with
+            % its corresponding template.
+            space = obj.space(space_name);
+            pmap = obj.probabilityMap(space.name);
+            template = space.getTemplate();
+
+            % to rgb
+            pmapRGB = cat(4, pmap, zeros(size(pmap)), zeros(size(pmap)));
+            templateRGB = cat(4, template, template, template);
+
+            % cutout
+            cutout = min(size(template), size(pmap));
+            pmapRGB = pmapRGB(1:cutout(1), 1:cutout(2), 1:cutout(3), :);
+            templateRGB = templateRGB(1:cutout(1), 1:cutout(2), 1:cutout(3), :);
+
+
+            % mix both layer
+            volume = pmapRGB .*0.5 + templateRGB;
+
+        end
+        function pmap = probabilityMap(obj, space_name)
+            found_space = false;
+            for i = 1:numel(obj.spaceAndRegionUrl.spaces)
+                if strcmp(obj.spaceAndRegionUrl.spaces(i).name, space_name)
+                    found_space = true;
+                    nifti_data = webread(obj.spaceAndRegionUrl.url(i));
+                    assert(strcmp(obj.spaceAndRegionUrl.spaces(i).format, 'nii'), "Currently supports nii format only")
+                    tmp_path = '+siibra/cache/tmp_nifti.nii.gz';
```

Comment:
I guess this leads to retrieving the file from the server each time it is requested. In typical workflows, probability maps will be reused often at runtime, so it might make sense to cache them to unique names.

 In Python, the probability maps of a parcellation are downloaded once, then converted into a memory-efficient sparse structure, and cached to disk persistently. This is something for a future extension, though.

Reply:
Agreed that caching would be a good direction, either short-term or mid-term. 

There are two levels:
* within session, which can be done with `memoize()`
* across sessions, which would need to be implemented likely using MAT file(s)

Reply:
Yes. In a first step I  added a cache of nifti files similar to what I've done for the space templates:
https://github.com/FZJ-INM1-BDA/siibra-matlab/pull/9/files#diff-9c017fa6292ee71613b308dd22d1c72e7584ed844193deeb22529007ddcddc80L22